compiler/damlc/tests/daml-test-files/Constraint.daml

```diff
@@ -2,7 +2,7 @@
 -- All rights reserved.
 
 {-# LANGUAGE ExistentialQuantification #-}
```

Comment:
huh, I didn’t know we support this and now I’m not sure we should.

Reply:
The type `Foo` below is not an existential type but to define it you need `ExistentialQuantification` or `GADTs` nevertheless. I think such a definition makes no sense at all but for some reason we decided to support. I agree that we should deprecate it. I'll have a look how it interacts with data-dependencies.

Reply:
In fact, we even have a check that the type looks like a record type since we need to inject more fields for the dictionaries: https://github.com/digital-asset/daml/blob/ac4fef2d44541687dfdba140cecf91519391e429/compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor.hs#L223
I really don't remember why we support this. But it makes zero sense to me. I don't see any reason why you would want to stuff the dictionary into the record rather than inject it whenever you call a typeclass method given that the `a` is universally quantified and not existential. 🤔

Reply:
Surprise! This does not work with data-dependencies. The error message I get is
```
  error:
  parse error on input ‘$’
  Perhaps you intended to use TemplateHaskell
Failed to compile interface for data-dependency: foo-0.0.1
```
To me, this looks like we're trying to put an identifier starting with a dollar symbol somewhere in the DAML file we're using to reconstruct the interface. Given that this feature is gated by either `ExistentialQuantification` or `GADTs`, which will both cause warnings about incompatibility with data-dependencies in the next SDK, I suggest we do exactly nothing about this. cc @sofiafaro-da